src/app/hooks/useGenerateVideo.ts

```diff
@@ -0,0 +1,45 @@
+import {useState} from 'react';
+
+export type dataForGenerationType = {
+	[key: string]: string | number | object | Date | undefined;
+};
+
+export const useGenerateVideo: (
+	data: dataForGenerationType,
+	compositionId: string,
+) => {
+	isLoading: boolean;
+	videoUrl: string | undefined;
+	error: string | undefined;
+	getVideoLink: () => void;
+} = (data, compositionId) => {
+	const [isLoading, setIsLoading] = useState(false);
+	const [videoUrl, setVideoUrl] = useState<string | undefined>(undefined);
+	const [error, setError] = useState<string | undefined>(undefined);
+
+	const getVideoLink = () => {
+		setIsLoading(true);
+		setError(undefined);
+
+		fetch(`https://api.shortvid.io/${compositionId}`, {
+			headers: {
+				'Content-Type': 'application/json',
+			},
+			method: 'POST',
+			body: JSON.stringify(data),
+		})
+			.then((res) => res.blob())
+			.then((blob) => {
+				const fileURL = URL.createObjectURL(blob);
+				setVideoUrl(fileURL);
+				setIsLoading(false);
+			})
+			.catch((err) => {
+				console.error(err);
+				setError('Error while generating the video');
+				setIsLoading(false);
+			});
+	};
```

Comment:
**suggestion:** We could avoid a lot of JS by using Basic form submission strategy.

Reply:
There are two compositions that requires a specific formatting for the sended props, so it will require a refacto of theses compositions, but i can test this solution. ✌️

Reply:
I tried something like that : 
![image](https://github.com/lyonjs/shortvid.io/assets/72607059/cf4f1148-b192-4b0e-bf61-38a73b85b208)

But it cause some issue/questions :
- By doing this, we send a post, but a get is automatically sent to retrieve the data, which causes an error.
- With this way of working, the page reloads and so we can't manage the loading for the user, the preventDefault would block the sending at all. I haven't looked at other possible solutions yet.
- By doing this, we'd be directly redirected to the video, which isn't necessarily good behaviour for the user.

Reply:
💡 
We can prevent the page load effect, I will let you dig up the spec for that.
There is a way to hook on the result of the form event if we use action :)